Spigot-API-Patches/0085-More-Anvil-Events.patch

```diff
@@ -0,0 +1,131 @@
+From 67a3105a61bf2e08f48f320a7c01a7967c0e6fdf Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <Blake.Galbreath@GMail.com>
+Date: Sun, 21 Jan 2018 05:09:37 -0600
+Subject: [PATCH] More Anvil Events
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/event/block/AnvilBreakEvent.java b/src/main/java/com/destroystokyo/paper/event/block/AnvilBreakEvent.java
+new file mode 100644
+index 00000000..0a5c8ed0
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/event/block/AnvilBreakEvent.java
+@@ -0,0 +1,40 @@
++package com.destroystokyo.paper.event.block;
++
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.inventory.InventoryEvent;
++import org.bukkit.inventory.AnvilInventory;
++import org.bukkit.inventory.InventoryView;
++
++/**
++ * Called when an anvil is broken from being used
++ */
++public class AnvilBreakEvent extends InventoryEvent implements Cancellable {
++    private static final HandlerList handlers = new HandlerList();
++    private boolean cancel;
++
++    public AnvilBreakEvent(InventoryView inventory) {
++        super(inventory);
++    }
++
++    @Override
++    public AnvilInventory getInventory() {
++        return (AnvilInventory) super.getInventory();
++    }
++
++    public boolean isCancelled() {
++        return cancel;
++    }
++
++    public void setCancelled(boolean cancel) {
++        this.cancel = cancel;
++    }
++
++    public HandlerList getHandlers() {
++        return handlers;
++    }
++
++    public static HandlerList getHandlerList() {
++        return handlers;
++    }
++}
+diff --git a/src/main/java/com/destroystokyo/paper/event/block/AnvilDamagedEvent.java b/src/main/java/com/destroystokyo/paper/event/block/AnvilDamagedEvent.java
+new file mode 100644
+index 00000000..7434513e
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/event/block/AnvilDamagedEvent.java
+@@ -0,0 +1,70 @@
++package com.destroystokyo.paper.event.block;
++
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.inventory.InventoryEvent;
++import org.bukkit.inventory.AnvilInventory;
++import org.bukkit.inventory.InventoryView;
++
++/**
++ * Called when an anvil is damaged from being used
++ */
++public class AnvilDamagedEvent extends InventoryEvent implements Cancellable {
++    private static final HandlerList handlers = new HandlerList();
++    private int damage;
++    private boolean cancel;
++
++    public AnvilDamagedEvent(InventoryView inventory, int damage) {
++        super(inventory);
++        this.damage = damage;
++    }
++
++    @Override
++    public AnvilInventory getInventory() {
++        return (AnvilInventory) super.getInventory();
++    }
++
++    /**
++     * Gets the anvil damage value
++     * <p>
++     * 0 = Not damaged
++     * 1 = Slightly damaged
++     * 2 = Very damaged
++     * 3+ = Anvil is breaking
```

Comment:
3 being the breaking point is an implementation detail.

I think we should consolidate the 2 events into this one, and add a isBreaking method that returns true if damage >= 3, and false if < 3.

Then we can add a setBreaking(boolean) that simply sets damage level to 3 if true, else leave as is if < 3, and set to 2 if was 3 before.